packages/core/src/carrier.ts

```diff
@@ -57,3 +57,25 @@ export function getSentryCarrier(carrier: Carrier): SentryCarrier {
   // rather than what's set in .version so that "this" SDK always gets its carrier
   return (__SENTRY__[SDK_VERSION] = __SENTRY__[SDK_VERSION] || {});
 }
+
+/**
+ * Returns a global singleton contained in the global `__SENTRY__[]` object.
+ *
+ * If the singleton doesn't already exist in `__SENTRY__`, it will be created using the given factory
+ * function and added to the `__SENTRY__` object.
+ *
+ * @param name name of the global singleton on __SENTRY__
+ * @param creator creator Factory function to create the singleton if it doesn't already exist on `__SENTRY__`
+ * @param obj (Optional) The global object on which to look for `__SENTRY__`, if not `GLOBAL_OBJ`'s return value
+ * @returns the singleton
+ */
+export function getGlobalSingleton<Prop extends keyof SentryCarrier>(
+  name: Prop,
+  creator: () => NonNullable<SentryCarrier[Prop]>,
+  obj = GLOBAL_OBJ,
+): NonNullable<SentryCarrier[Prop]> {
+  const __SENTRY__ = (obj.__SENTRY__ = obj.__SENTRY__ || {});
+  const carrier = (__SENTRY__[SDK_VERSION] = __SENTRY__[SDK_VERSION] || {});
```

Comment:
actually no 😬 I had that first but it broke loader tests. This is why I added the comment below - if we'd call that, it would already set the version, which would lead to stuff thinking the SDK is set up, when it isn't. We store e.g. the default global/isolation scope on there even before the SDK is setup, so we need to ensure the `version` is not set here yet!

Reply:
(on the plus side, the loader tests are good! xD)

Reply:
ah sorry, I should have read the comment 🤦 thanks!